nova_vm/src/engine/bytecode/executable.rs

```diff
@@ -1171,84 +1171,76 @@ impl CompileEvaluation for ast::ArrayExpression<'_> {
     }
 }
 
-impl CompileEvaluation for ast::Argument<'_> {
-    fn compile(&self, ctx: &mut CompileContext) {
-        match self {
-            ast::Argument::SpreadElement(_) => {
-                panic!("Cannot support SpreadElements currently")
+fn compile_arguments(arguments: &[ast::Argument], ctx: &mut CompileContext) -> usize {
+    // If the arguments don't contain the spread operator, then we can know the
+    // number of arguments at compile-time and we can pass it as an argument to
+    // the call instruction.
+    // Otherwise, the first time we find a spread operator, we need to start
+    // tracking the number of arguments in the compiled bytecode. We store this
+    // number in the result value, and we pass u16::MAX to the call instruction.
+    let mut known_num_arguments = Some(0 as IndexType);
+
+    for argument in arguments {
+        // If known_num_arguments is Some, the stack contains the number of
```

Comment:
**nitpick:** Isn't this the other way around: If it's `None` then we've already loaded the number of arguments on the stack, if it's Some then we've yet to do that.

Reply:
Fixed.